bundles/org.openhab.binding.mideaac/src/main/java/org/openhab/binding/mideaac/internal/connection/ConnectionManager.java

```diff
@@ -0,0 +1,597 @@
+    private Response lastResponse;
+    private CloudProvider cloudProvider;
+    private Security security;
+    private final int version;
+    private final boolean promptTone;
+    private boolean deviceIsConnected;
+    private int droppedCommands = 0;
+
+    /**
+     * True allows command resend if null and timeout response
+     */
+    private boolean resend = true;
+
+    /**
+     * Connection manager configuration
+     * 
+     * @param ipAddress Device IP
+     * @param ipPort Device Port
+     * @param timeout Socket timeout
+     * @param key Security key V.3
+     * @param token Security token V.3
+     * @param cloud Cloud Provider
+     * @param email Cloud Provider login email
+     * @param password Cloud Provider login password
+     * @param deviceId Device ID
+     * @param version Device version
+     * @param promptTone Tone after command true or false
+     */
+    public ConnectionManager(String ipAddress, int ipPort, int timeout, String key, String token, String cloud,
+            String email, String password, String deviceId, int version, boolean promptTone) {
+        this.deviceIsConnected = false;
+        this.ipAddress = ipAddress;
+        this.ipPort = ipPort;
+        this.timeout = timeout;
+        this.key = key;
+        this.token = token;
+        this.cloud = cloud;
+        this.deviceId = deviceId;
+        this.version = version;
+        this.promptTone = promptTone;
+        this.lastResponse = new Response(HexFormat.of().parseHex("C00042667F7F003C0000046066000000000000000000F9ECDB"),
+                version);
+        this.cloudProvider = CloudProvider.getCloudProvider(cloud);
+        this.security = new Security(cloudProvider);
+    }
+
+    private Socket socket = new Socket();
+    private InputStream inputStream = new ByteArrayInputStream(new byte[0]);
+    private DataOutputStream writer = new DataOutputStream(System.out);
+
+    /**
+     * Gets last response
+     * 
+     * @return byte array of last response
+     */
+    public Response getLastResponse() {
+        return this.lastResponse;
+    }
+
+    /**
+     * The socket is established with the writer and inputStream (for reading responses)
+     * V2 devices will proceed to send the poll or the set command.
+     * V3 devices will proceed to authenticate
+     * 
+     * @throws MideaConnectionException
+     * @throws MideaAuthenticationException
+     * @throws SocketTimeoutException
+     * @throws IOException
+     */
+    public synchronized void connect()
+            throws MideaConnectionException, MideaAuthenticationException, SocketTimeoutException, IOException {
+        logger.trace("Connecting to {}:{}", ipAddress, ipPort);
+
+        int maxTries = 3;
+        int retrySocket = 0;
+
+        // If resending command add delay to avoid connection rejection
+        // Suspect that the AC device needs a few seconds to clear.
+        if (!resend) {
+            try {
+                Thread.sleep(5000);
```

Comment:
Noticed two things:
1. the condition seems the reverse. currently it sleeps when not resending
2. a hardcoded sleep is a code smel, what other options did you consider?
3. both a synchronized method and a long (5 second) sleep are error prone.

Reply:
This one was left open.

Reply:
My point was that i expect the sleep when (resend == true), but the current condition is the opposite (resend == false)

Reply:
Reviewed. If the resend is true (as it normally is) and the response bytes are still null it hits this `else` (line 401) to start the retry.
```
            } else {
                if (resend) {
                    logger.debug("Resending Command {}", command);
                    resend = false;
                    sendCommand(command, callback);
```
The resend is set to false, so I needed a delay before it tries to create a new socket. As noted elsewhere, I had trouble with an immediate resend. The resend doesn't turn to true again until a successful poll (line 359)
```
            if (responseBytes != null) {
                resend = true;
```